src/mscorlib/src/System/String.cs

```diff
@@ -851,14 +851,12 @@ private unsafe String CtorCharPtrStartLength(char *ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         internal extern bool TryGetTrailByte(out byte data);
 
-#if !FEATURE_CORECLR
         public CharEnumerator GetEnumerator() {
             Contract.Ensures(Contract.Result<CharEnumerator>() != null);
             Contract.EndContractBlock();
             BCLDebug.Perf(false, "Avoid using String's CharEnumerator until C# special cases foreach on String - use the indexed property on String instead.");
```

Comment:
What does this Perf call do?  Is it an assert?  I'm wondering if we need to remove it.